anyio/__init__.py

```diff
@@ -292,34 +299,65 @@ def aopen(file: Union[str, 'os.PathLike', int], mode: str = 'r', buffering: int
         :exc:`~ssl.SSLEOFError` may be raised during reads from the stream.
         Some protocols, such as HTTP, require this option to be ``False``.
         See :meth:`~ssl.SSLContext.wrap_socket` for details.
+    :param happy_eyeballs_delay: delay (in seconds) before starting the next connection attempt
     :return: a socket stream object
+    :raises OSError: if the connection attempt fails
 
     """
+    # Placed here due to https://github.com/python/mypy/issues/7057
+    stream = None  # type: Optional[SocketStream]
+
+    async def try_connect(af: int, addr: str, delay: float):
+        nonlocal stream
+
+        if delay:
```

Comment:
Unconditional delay instead of launching all tasks at once means we don't continue to the next attempt if one fails immediately.

Reply:
I'm not sure what you mean. This check only returns false for the first attempt. I wanted to eliminate the dummy `sleep(0)` this way.

Reply:
There's no way to signal to the next task that "we failed, go ahead and start even if you're still waiting".

Reply:
Right, now I get you. I'll fix this.

Reply:
The latest commit fixes this.